main.tf

```diff
@@ -15,47 +19,64 @@ resource "aws_rds_cluster" "this" {
   vpc_security_group_ids          = concat([aws_security_group.this.id], var.additional_security_groups)
   tags                            = var.tags
   db_cluster_parameter_group_name = var.db_cluster_parameter_group_name
-  deletion_protection             = true
-}
-
-resource "aws_secretsmanager_secret" "root_password" {
-  name_prefix = "aurora-root-${var.name}"
-  description = "Root password for the ${var.name} aurora cluster database"
-  tags        = var.tags
-}
-
-resource "aws_secretsmanager_secret_version" "root_password" {
-  secret_id     = aws_secretsmanager_secret.root_password.id
-  secret_string = random_password.password.result
-}
+  deletion_protection             = var.deletion_protection
+  copy_tags_to_snapshot           = true
 
-resource "random_password" "password" {
-  length           = 32
-  special          = true
-  min_special      = 1
-  override_special = "-._~" # URL-safe characters prevent parsing errors when using this password in a connection string
+  enabled_cloudwatch_logs_exports = [
+    "postgresql",
+  ]
 }
 
 resource "aws_secretsmanager_secret" "connection_string" {
+  # checkov:skip=CKV2_AWS_57: RDS connection strings cannot be rotated
+  # checkov:skip=CKV_AWS_149: We will use AWS managed keys because CMK are expensive and not necessary for our use case
   name_prefix = "aurora-connectionstring-${var.name}"
   description = "Connection String for the ${var.name} aurora cluster database"
   tags        = var.tags
 }
 
+data "aws_secretsmanager_secret_version" "root_password" {
+  secret_id = aws_rds_cluster.this.master_user_secret[0].secret_arn
+}
+
 resource "aws_secretsmanager_secret_version" "connection_string" {
   secret_id     = aws_secretsmanager_secret.connection_string.id
-  secret_string = "postgresql://${aws_rds_cluster.this.master_username}:${aws_secretsmanager_secret_version.root_password.secret_string}@${aws_rds_cluster.this.endpoint}:${aws_rds_cluster.this.port}/${aws_rds_cluster.this.database_name}"
+  secret_string = "postgresql://${aws_rds_cluster.this.master_username}:${data.aws_secretsmanager_secret_version.root_password.secret_string}@${aws_rds_cluster.this.endpoint}:${aws_rds_cluster.this.port}/${aws_rds_cluster.this.database_name}"
```

Comment:
Just found out the secret the managed secret generates is actually the username & password in a json format like:
```
{
username: root
password: <password>
}
```

instead of just the password. We'll need to handle that appropriately.

Reply:
Fixed in commit [7875629](https://github.com/synapsestudios/terraform-aws-rds-aurora-cluster/pull/3/commits/78756293cc4b2eda78c79bf7550e70ffa8f30175)